src/libraries/Common/src/System/Security/Cryptography/ECCng.ImportExport.cs

```diff
@@ -152,43 +118,27 @@ internal static byte[] GetPrimeCurveBlob(ref ECParameters parameters, bool ecdh)
 
         internal static void ExportNamedCurveParameters(ref ECParameters ecParams, byte[] ecBlob, bool includePrivateParameters)
         {
-            // We now have a buffer laid out as follows:
-            //     BCRYPT_ECCKEY_BLOB   header
-            //     byte[cbKey]          Q.X
-            //     byte[cbKey]          Q.Y
-            //     -- Private only --
-            //     byte[cbKey]          D
-
-            KeyBlobMagicNumber magic = (KeyBlobMagicNumber)BitConverter.ToInt32(ecBlob, 0);
+            ECParameters temp = default;
 
-            // Check the magic value in the key blob header. If the blob does not have the required magic,
-            // then throw a CryptographicException.
-            CheckMagicValueOfKey(magic, includePrivateParameters);
-
-            unsafe
-            {
-                // Fail-fast if a rogue provider gave us a blob that isn't even the size of the blob header.
-                if (ecBlob.Length < sizeof(BCRYPT_ECCKEY_BLOB))
-                    throw ErrorCode.E_FAIL.ToCryptographicException();
-
-                fixed (byte* pEcBlob = &ecBlob[0])
+            DecodeEccKeyBlob(
+                ecBlob,
+                (KeyBlobMagicNumber magic, byte[] x, byte[] y, byte[]? d) =>
```

Comment:
```suggestion
                static (KeyBlobMagicNumber magic, byte[] x, byte[] y, byte[]? d) =>
```

If possible.  And if not... try to make it possible :smile: